packages/gherkindoc/src/cmd/xlsx.ts

```diff
@@ -0,0 +1,321 @@
+    ws.lastRow.font = styles.light;
+  });
+
+  if (step.dataTable) {
+    genDataTable(ws, step.dataTable, testers || 0, maxWidths);
+  }
+};
+
+function genExampleTable(
+  ws: any,
+  example: any,
+  testers: number,
+  maxWidths: any,
+) {
+  example.beforeComments.forEach((comment: any) => {
+    ws.addRow({content0: comment});
+    ws.lastRow.font = styles.light;
+  });
+
+  ws.addRow({content0: 'Examples:'});
+  const headerRow: any = {};
+  const cellsToStyle: any[] = [];
+
+  example.header.forEach((header: any, index: any) => {
+    const key = `content${index + 2}`;
+    headerRow[key] = header;
+    if (maxWidths[key] < header.toString().length) {
+      maxWidths[key] = header.toString().length;
+    }
+    cellsToStyle.push(index + 4 + testers);
+  });
+
+  ws.addRow(headerRow);
+  ws.lastRow.font = styles.boldItalic;
+
+  cellsToStyle.forEach(cellNum => {
+    ws.lastRow.getCell(cellNum).fill = styles.blueFill;
+    ws.lastRow.getCell(cellNum).border = styles.fullBorder;
+  });
+
+  example.data.forEach((row: any) => {
+    const cellsToStyle: any[] = [];
+    const contentRow: any = {};
+
+    row.forEach((entry: any, index: any) => {
+      const key = `content${index + 2}`;
+      contentRow[key] = entry;
+
+      if (maxWidths[key] < entry.toString().length) {
+        maxWidths[key] = entry.toString().length;
+      }
+
+      cellsToStyle.push(index + 4 + testers);
+    });
+
+    ws.addRow(contentRow);
+
+    cellsToStyle.forEach(cellNum => {
+      ws.lastRow.getCell(cellNum).border = styles.fullBorder;
+    });
+  });
+
+  example.afterComments.forEach((comment: any) => {
+    ws.addRow({content0: comment});
+    ws.lastRow.font = styles.light;
+  });
+}
+
+function genDataTable(ws: any, table: any[], testers: number, maxWidths: any) {
+  if (table.length === 0) {
+    return;
+  }
+
+  const headerRow: any = {};
+  const cellsToStyle: any[] = [];
+
+  table[0].forEach((header: any, index: any) => {
+    const key = `content${index + 2}`;
+    headerRow[key] = header;
+
+    if (maxWidths[key] < header.toString().length) {
+      maxWidths[key] = header.toString().length;
+    }
+
+    cellsToStyle.push(index + 4 + testers);
+  });
+
+  ws.addRow(headerRow);
+  ws.lastRow.font = styles.boldItalic;
+
+  cellsToStyle.forEach(cellNum => {
+    ws.lastRow.getCell(cellNum).fill = styles.blueFill;
+    ws.lastRow.getCell(cellNum).border = styles.fullBorder;
+  });
+
+  table.shift();
+  table.forEach((row: any) => {
+    const cellsToStyle: any[] = [];
+    const contentRow: any = {};
+
+    row.forEach((entry: any, index: any) => {
+      const key = `content${index + 2}`;
+      contentRow[key] = entry;
+
+      if (maxWidths[key] < entry.toString().length) {
+        maxWidths[key] = entry.toString().length;
+      }
+
+      cellsToStyle.push(index + 4 + testers);
+    });
+
+    ws.addRow(contentRow);
+
+    cellsToStyle.forEach(cellNum => {
+      ws.lastRow.getCell(cellNum).border = styles.fullBorder;
+    });
+  });
+}
```

Comment:
Would be good to include some inline-comments/documentation in here for future developers